paper.md

```diff
@@ -0,0 +1,115 @@
+
+In contrast, the multi-node setup exhibits less ideal scaling \autoref{fig:dist} with a measured speedup approximately 55% that of perfect scaling. We attribute this scaling behavior to hardware-dependent network communication overhead. Distributing a `sleep(5)` function allocated 6 CPU cores per task (to mimic a fairly quick docking simulation) in parallel over differing hardware setups
+led to an approximate 2.5% increase in wall-time relative to the single-node setup each time the number of nodes in the setup was doubled while keeping the total number of CPU cores the same. Such a trend is consistent with network communication being detrimental to scaling behavior. This test also communicated the absolute minimum amount of data over the network, as there were no function arguments or return values. When communicating `CalculationData` objects (approximately 600 bytes in serialized form) over the network, as in `pyscreener`, the drop increased to 6% for each doubling of the total number of nodes. Minimizing the total size of `CalculationData` objects was therefore an explicit implementation goal. Future development will seek to further reduce network communication overhead costs to bring `pyscreener` scaling closer to ideal scaling.
+
+# Examples
+To illustrate `pyscreener`, we consider docking benezene (SMILES string `"c1ccccc1"`) against 5WIU with a docking box centered at (-18.2, 14.4, -16.1) with x-, y-, and z-radii (15.4, 13.9, 14.5). We may perform this docking using AutoDock Vina over 6 CPU cores via `pyscreener` like so:
+```python
+>>> import pyscreener as ps
+>>> metadata = ps.build_metadata("vina")
+>>> virtual_screen = ps.virtual_screen(
+...   "vina", receptors=["5WIU.pdb"],
+...   center=(-18.2, 14.4, -16.1),
+...   size=(15.4, 13.9, 14.5),
+...   metadata_template=metadata,
+...   ncpu=6
+... )
+>>> scores = virtual_screen("c1ccccc1")
+>>> scores
+array([-4.4])
+```
+
+Alternatively, we may dock many molecules by passing a `List` of SMILES strings to the `DockingVirtualScreen`:
+```python
+>>> smis = [
+...   "c1ccccc1",
+...   "O=C(Cc1ccccc1)NC1C(=O)N2C1SC(C2C(=O)O)(C)C",
+...   "C=CCN1CCC23C4C(=O)CCC2(C1CC5=C3C(=C(C=C5)O)O4)O"
+... ]
+>>> scores = virtual_screen(smis)
+>>> scores.shape
+(3,)
+```
+
+By default, AutoDock Vina docks molecules using an `--exhaustiveness` value of 8, but we may specify a higher number in the `metadata`:
+```python
+>>> metadata = ps.build_metadata("vina", dict(exhaustivness=32))
+```
+We may also utilize other docking engines in the AutoDock Vina family by specifying the `software` for Vina-type metadata. Here, we use the accelerated optimization routine of QVina for faster docking. Note that we also support `software` values of `"smina"` and `"psovina"` in addition to `"vina"` and `"qvina"`.
```

Comment:
Could you add links/references to the different software packages listed here?
cc https://github.com/openjournals/joss-reviews/issues/3950

Reply:
@mikemhenry references have been added!